### PR TITLE
Require Entitlements.HA_ADMIN for HA clear requests; tidy up

### PR DESCRIPTION
**Pull request by algairim**
Signed-off-by: Mykola Mandra <mykola.mandra@cloudsoft.io>

### CODE REVIEW

File: rest/rest-api/src/main/java/org/apache/brooklyn/rest/api/ServerApi.java [hunk 1]
```diff
@@ -163,7 +163,7 @@ public Response clearHighAvailabilityPlaneStates(
     @GET
     @Path("/ha/priority")
     @ApiOperation(value = "Returns the HA node priority for MASTER failover")
-    public long getHighAvailabitlityPriority();
+    public long getHighAvailabilityPriority();
```
Comment (algairim) [inline rest/rest-api/src/main/java/org/apache/brooklyn/rest/api/ServerApi.java:166]: Typo in the name.

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ServerResource.java [hunk 2]
```diff
@@ -494,16 +493,16 @@ public HighAvailabilitySummary getHighAvailabilityPlaneStates() {
 
     @Override
     public Response clearHighAvailabilityPlaneStates() {
-        if (!Entitlements.isEntitled(mgmt().getEntitlementManager(), Entitlements.SYSTEM_ADMIN, null))
-            throw WebResourceUtils.forbidden("User '%s' is not authorized to perform this operation", Entitlements.getEntitlementContext().user());
+        if (!Entitlements.isEntitled(mgmt().getEntitlementManager(), Entitlements.HA_ADMIN, null))
```
Comment (algairim) [inline rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ServerResource.java:496]: Require `Entitlements.HA_ADMIN` to clear HA.

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ServerResource.java [hunk 3]
```diff
@@ -494,16 +493,16 @@ public HighAvailabilitySummary getHighAvailabilityPlaneStates() {
...
+            throw WebResourceUtils.forbidden(USER_OPERATION_NOT_AUTHORIZED_MSG, Entitlements.getEntitlementContext().user());
         mgmt().getHighAvailabilityManager().publishClearNonMaster();
         return Response.ok().build();
     }
 
     @Override
     public Response clearHighAvailabilityPlaneStates(String nodeId) {
-        if (!Entitlements.isEntitled(mgmt().getEntitlementManager(), Entitlements.SYSTEM_ADMIN, null))
-            throw WebResourceUtils.forbidden("User '%s' is not authorized to perform this operation", Entitlements.getEntitlementContext().user());
+        if (!Entitlements.isEntitled(mgmt().getEntitlementManager(), Entitlements.HA_ADMIN, null))
```
Comment (algairim) [inline rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ServerResource.java:504]: Require `Entitlements.HA_ADMIN` to clear HA.

**Comment by jcabrerizo** [comment]
LGTM @algairim thanks!

**Pull request merged**